lib/checks/tables/td-headers-attr-evaluate.js

```diff
@@ -14,9 +14,17 @@ export default function tdHeadersAttrEvaluate(node) {
     }
   }
 
-  const ids = cells
-    .filter(cell => cell.getAttribute('id'))
-    .map(cell => cell.getAttribute('id'));
+  // Collect table headers
+  // Get a list all the values of the headers attributes
+  // Find header cells inside the table
+  // Filter td elements and map to ids
+  const tableHeaders = cells
+    .filter(cell => cell.getAttribute('headers'))
+    .flatMap(cell => tokenList(cell.getAttribute('headers').trim()))
+    .map(headerId => cells.find(cell => cell.getAttribute('id') === headerId))
+    .filter(headerCell => headerCell)
+    .filter(headerCell => headerCell.nodeName.toLowerCase() !== 'td')
+    .map(headerCell => headerCell.getAttribute('id'));
```

Comment:
i think there is an unnecessary amount of looping in here. It's probably better to stick more closely to what this was before and add the node name to it. I also think this is going to need to consider roles. An td with role=columnheader/rowheader should probably still pass.